src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts

```diff
@@ -993,3 +995,15 @@ export const handleGetMetadata = [
   }),
   getMetadata,
 ] as ControllerHandler[]
+
+export const handleGetS3PresignedPostData: ControllerHandler<
+  unknown,
+  Record<string, S3.PresignedPost> | ErrorDto,
+  Record<string, number>
```

Comment:
likewise - could we be more specific about the type of the request body, and enforce joi validation on the key names?

Reply:
Oh wait, I realise its because you expect the Dto to be
{<fieldName>: <contentLength>}[]

are you still able to apply joi validation if <fieldName> is a key and not a value?

Reply:
Did a bit of research and I think keys will always be strings by the nature of JSONs. The level of validation we can perform on keys are limited though, like in this case I can't regex test it. I'll refactor to it to your suggested structure so we have better validation later today or tomorrow 👍 Thanks for flagging it!

Reply:
Stricter joi validation enforced in [447b6ab](https://github.com/opengovsg/FormSG/pull/6685/commits/447b6abfcc83cf79ee8ca232155df9bb0a15fde7) 👍 Thanks! :)